packages/switch/src/ux-switch.ts

```diff
@@ -0,0 +1,112 @@
+import { customElement, bindable, ViewResources } from 'aurelia-templating';
+import { computedFrom, bindingMode } from 'aurelia-binding';
+import { inject } from 'aurelia-dependency-injection';
+import { StyleEngine, UxComponent, normalizeBooleanAttribute } from '@aurelia-ux/core';
+import { UxSwitchTheme } from './ux-switch-theme';
+
+@inject(Element, ViewResources, StyleEngine)
+@customElement('ux-switch')
+
+export class UxSwitch implements UxComponent {
+  @bindable public disabled: boolean | string = false;
+  @bindable public effect = null;
+  @bindable public id: string;
+  @bindable public tabindex = 0;
+  @bindable public theme: UxSwitchTheme;
+
+  @bindable({ defaultBindingMode: bindingMode.twoWay })
+  @bindable public checked: boolean = false;
+
+  @computedFrom('disabled')
+  public get isDisabled() {
+    return normalizeBooleanAttribute('disabled', this.disabled);
+  }
+
+  constructor(public element: HTMLElement, public resources: ViewResources, private styleEngine: StyleEngine) {
+    styleEngine.ensureDefaultTheme(new UxSwitchTheme());
```

Comment:
Weren't we going to move this to an external shared theme instance?

Reply:
We looked at it but if I remember correctly, the CSS variables made that a moot point.

Right now when a component is created, UX will look for a registered default theme. If a default theme does not exist, this creates a new instance at the root with the appropriate CSS variables. If those CSS variables are changed, it immediately cascades to all controls in the document that use that CSS variable, aka the components that use the theme. This is unless the component is bound to a different theme, in which case it is using a set of CSS variables from that bound theme instead which overrides the default.

Reply:
I thought we were using a shared instance just to avoid the unnecessary class instantiation, esp. in the case where there may be lots of components.